### DISCUSSION TITLE
Any demo/examples for xterm-headless?

### DISCUSSION THREAD

**Discussion by alanhe421** [sourcelair/xterm.js#4870] [Q&A]
I'm not sure what headless does and how to use it.

**Comment by InTEGr8or** [comment]
According to the https://www.npmjs.com/package/@xterm/headless

> xterm-headless is a headless terminal that can be run in node.js. This is useful in combination with the frontend [xterm](https://www.npmjs.com/package/xterm) for example to keep track of a terminal's state on a remote server where the process is hosted.

It also indicates that it is the similar to `@xterm/xterm` except without reference to DOM elements:

> Addons in xterm-headless work the [same as in xterm](https://github.com/xtermjs/xterm.js/blob/master/README.md#addons) with the one caveat being that the addon needs to be packaged for node.js and not use any DOM APIs.

VS Code uses the `@xterm/headless`, which provides a type of example:

https://github.com/microsoft/vscode/blob/366fac3cfdbdf949a33678ed7da05faea0bbfec5/src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts#L21
